main.go

```diff
@@ -28,7 +29,27 @@ var (
 	}
 )
 
+// We supply some context from API Gateway to decide which signing key to use.
+// This could easily be extended to contain additional context.
+type APIGatewayContext struct {
+	SourceIP string
+	UserArn  string
+}
+
+// We just implement a DefaultSigner that always uses HALLOW_KMS_KEY_ARN.
+// But you could implement your own SignerChooser that uses APIGatewayContext.
+type DefaultSigner struct {
+	DefaultKMSKeyARN string
+}
+
+func (d DefaultSigner) Choose(context APIGatewayContext) (crypto.Signer, error) {
+	sess := session.New()
+	return kmssigner.New(kms.New(sess), d.DefaultKMSKeyARN)
```

Comment:
I think we should reuse the `kmssigner` instance, constructing new AWS sessions is sloooow